lib/identity/client.js

```diff
@@ -205,6 +206,62 @@ class Client extends PartialClient {
     const newSecretCreated = this.createSecret(newSecret)
     return newSecretCreated
   }
+  /**
+   * shareSecretWithUsername
+   *
+   * @param {String} secretType
+   * @param {String} secretName
+   * @param {Array} usernameToAdd
+   *
+   *
+   */
+  async shareSecretWithUsername(secretName, secretType, usernamesToAdd = []) {
+    // Generate Group to Validate it exists
+    const groupName = `tozny.secret.${this.config.realmName}.${
+      this.storage.config.clientId
+    }.${secretType.toLowerCase()}`
+    const groupList = await this.storage.listGroups(
+      this.storage.config.clientId,
+      [groupName]
+    )
+    if (groupList.groups.length < 1) {
+      throw new Error('Group Does Not Exist')
+    }
+    const currentGroup = groupList.groups[0]
+    // Create the request for username to client id map
+    const SearchRealmIdentitiesRequest = {
```

Comment:
Doesn't this already have an sdk method? If not it should be extracted out.

Reply:
nope! just in the identity portal! but ill make it now!